Downloader UI cleanup and icons

comma opened the pull request:
Cleans stuff up a bit, adds icons.

![](https://vgy.me/37IDRJ.png)

![](https://vgy.me/VuX2Nc.png)

comma committed:
Cleans up soft downloader list

Condenses it in a table, doesn't list filename.

comma committed:
Adds program icons

They show up in main menu and in downloader.
Revelation program tweaked so if you name it after some existing program it tweaks its icon after it. Not that it's ever used but why not.

ghost commented:
The icons are slightly off-center on the lower image. Is this intentional?

atlantiscze commented on nano/templates/laptop_mainscreen.tmpl:
```diff
@@ -1,8 +1,7 @@
 {{for data.programs}}
-	<tr><td>{{:helper.link(value.desc, null, {'PC_runprogram' : value.name})}}
-	<td>{{:helper.link('X', null, {'PC_killprogram' : value.name}, (value.running ? null : 'disabled'))}}
-	<td>{{:helper.link('AR', null, {'PC_setautorun' : value.name}, null, (value.autorun ? 'yellowButton' : null))}}
```
You probably forgot to re-add this, at this state it removes the autorun feature from the menu.

comma replied:
No I deliberately removed it as it's just a clutter and an unused feature.

atlantiscze replied:
"Unused feature" is your subjective opinion. It is being used, and it is preset on some computers, therefore it should be controllable from the UI. Nobody is forcing you to click the button after all.

comma replied:
w/e

atlantiscze commented on nano/templates/ntnet_downloader.tmpl:
```diff
@@ -57,64 +57,21 @@
-				{{:helper.link("DOWNLOAD", null, {'PRG_downloadfile' : value.filename})}}
-			</div>	
+		<tr><td><td>{{:helper.link('', value.icon, {'PRG_downloadfile' : value.filename})}}	
+			<td><span class='white'>{{:value.filedesc}}</span>
```
Information about file size is gone, it should probably be added back.

EDIT: When i look at the UI, it might be good to add this information to the button that triggers the download. That would also resolve @Rebruiza 's question.

comma replied:
I tried that, but the buttons don't get linebreaks, so programs with long names wreck the layout.

atlantiscze replied:
How about adding it (concat-ing) it into the filedesc field then? It should fit there quite well as it's already designed for longer texts.

comma replied:
I am not sure what you mean, if you mean size I just readded it back as a table cell, might as well have it in case at some point disk space actually matters.

atlantiscze reviewed (changes requested):
Overall majority of the changes look good, there is just few missing things that should be re-added.

comma committed:
Readds extremely useful and widely used feature

Now those numberous players that used it aside from seeing yellow button on preset computers will be saved.
Also random empty cell gone

atlantiscze reviewed (dismissed):
Feedback has been addressed :+1: && travis()

Techhead0 commented on nano/templates/ntnet_downloader.tmpl:
```diff
@@ -57,64 +57,23 @@
-			<div class="itemContent">
-				{{:helper.link("DOWNLOAD", null, {'PRG_downloadfile' : value.filename})}}
-			</div>	
+		<tr><td><td>{{:helper.link('', value.icon, {'PRG_downloadfile' : value.filename})}}	
```
Empty table cell at the start of each row?

comma replied:
Whoops, it's not visible I guess.

Techhead0 commented on nano/templates/ntnet_downloader.tmpl:
```diff
@@ -57,64 +57,23 @@
-					{{:helper.link("DOWNLOAD", null, {'PRG_downloadfile' : value.filename})}}
-				</div>	
-			</div>
+		<tr><td><td>{{:helper.link(value.filedesc, value.icon, {'PRG_downloadfile' : value.filename})}}	
```
Should these table elements be inside a table?

comma replied:
I guess since they're all same sized it would work either way.

comma replied:
Actually they should, since icon per row.

Techhead0 reviewed (commented):
Few HTML things.
